File: src/P021/p021.py
from src.utils.py.math import proper_divisors


def is_amicable(num: int) -> bool:
    divisor_sum = sum(proper_divisors(num))
    return divisor_sum != num and sum(proper_divisors(divisor_sum)) == num


def solve() -> int:
    return sum(num for num in range(2, 10001) if is_amicable(num))

File: src/P021/p021.go
package problem021

import (
	"strconv"

	"github.com/sunilbpandey/project-euler/src/utils/go/math/intutils"
)

func isAmicable(n int) bool {
	m := intutils.SumOfProperDivisors(n)
	return m != n && intutils.SumOfProperDivisors(m) == n
}

func Solve() string {
	sum := 0
	for n := 2; n < 10000; n++ {
		if isAmicable(n) {
			sum += n
		}
	}
	return strconv.Itoa(sum)
}
